test/tests.spec.ts

```diff
@@ -571,6 +571,31 @@ describe("node-saml /", function () {
         expect(metadata).to.contain('WantAssertionsSigned="true"');
       });
 
+      it("generateServiceProviderMetadata throw error", function () {
+        const samlConfig: SamlConfig = {
+          cert: TEST_CERT,
+          issuer: "http://example.serviceprovider.com",
+          callbackUrl: "http://example.serviceprovider.com/saml/callback",
+          identifierFormat: "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
+          decryptionPvk: fs.readFileSync(__dirname + "/static/testshib encryption pvk.pem"),
+          wantAssertionsSigned: true,
+        };
+
+        const samlObj = new SAML(samlConfig);
+        let metadata;
+        try {
+          metadata = samlObj.generateServiceProviderMetadata(null);
+        } catch (error) {
+          // typescript
+          if (error instanceof Error) {
+            expect(error.toString()).to.contain(
+              "Error: Missing decryptionCert while generating metadata for decrypting service provider"
+            );
+          }
+        }
+        expect(metadata).to.be.undefined;
+      });
+
```

Comment:
Additional test coverage is welcome, but what does this have to with the change from `url.parse() to `new URL()`?

Reply:
Its a legacy API

The Node.sj documentation said `Use the WHATWG URL API instead.`

- [https://nodejs.org/api/url.html#urlparseurlstring-parsequerystring-slashesdenotehost](https://nodejs.org/api/url.html#urlparseurlstring-parsequerystring-slashesdenotehost)

The possible argument tu use `url.parse` are :
- `Faster than the alternative WHATWG URL parser.` - it is used in `test` files so I think it's not the purpose here
- `Easier to use with regards to relative URLs than the alternative WHATWG URL API.` - the `new URL` works good in our case

Reply:
But I agree it can be split into 2 PR

Reply:
I'm not concerned about the performance difference in the URL parses. SAML would a small amount of traffic for most web-apps and URL parsing is probably not the bottleneck. 

I'm OK with leaving this as one PR as long as @cjbarth agrees about accepting the additional test coverage.

Reply:
Okay great

I can also split the PR if needed 😃